### PR TITLE
Add nullability annotations

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Adds `@Nonnull` and `@Nullable` annotations for explicit nullability. This is also very useful for kotlin which makes use of these annotations in its type system.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/entities/Invite.java [hunk 1]
```diff
@@ -80,6 +82,7 @@
      * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.api.entities.Invite Invite}
      *         <br>The Invite object
      */
+    @Nonnull
     static RestAction<Invite> resolve(final JDA api, final String code, final boolean withCounts)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Invite.java]: Seems I forgot to add annotations to parameters here

**Comment by MinnDevelopment** [comment]
We should consider changing how we handle conditional nullability in types such as `Message`. For instance:

```java
if (message.isFromType(ChannelType.TEXT)) {
    TextChannel channel = message.getTextChannel();
    channel.getManager().setTopic("text").queue();
}
```

In this code an IDE will warn about `channel` being possibly null, which should be impossible when checking for the type beforehand. This can be avoided by throwing an exception instead of returning null. Opinions?

File: src/main/java/net/dv8tion/jda/api/entities/Activity.java [hunk 2]
```diff
@@ -47,13 +48,15 @@
...
+    @Nullable
     RichPresence asRichPresence();
 
     /**
      * The displayed name of the {@link Activity Activity}. If no name has been set, this returns null.
      *
-     * @return Possibly-null String containing the Activity's name.
+     * @return String containing the Activity's name.
      */
+    @Nonnull
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/entities/Activity.java]: This was originally `Possibly-null` but the `Notnull` annotation was add. Is that correct?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/entities/Activity.java]: The `EntityBuilder` uses `String.valueOf(json.get("name"))` so this can't be null.

**Comment by DV8FromTheWorld** [comment]
> This can be avoided by throwing an exception instead of returning null. Opinions?

Throwing instead of returning `null` for things like `getTextChannel` when the message did not come from a `TextChannel` seems completely appropriate. I would not be against this change as it shows to the developer, immediately, what the problem is instead of receiving an NPE.

I wonder how many people have code like this though:
```
TextChannel tchan = message.getTextChannel();
if (tchan != null) {
  // ... do a thing 
}
```
Just curious.

I would go ahead with making the channel accessor methods on Message / Message Events throw.

**Comment by DV8FromTheWorld** [comment]
Also, removing the proposal tag as this is good. Lets include this.

**Pull request merged**